### ISSUE TITLE
New service `Type=any`

### ISSUE THREAD

**Issue by unusual-thoughts** [systemd/systemd#16805]
**Is your feature request related to a problem? Please describe.**
As part of moving desktop applications to be managed by systemd as transient services in KDE, while most `.desktop` apps have the parent process run for the lifetime of the application (corresponding to `Type=simple`), some apps fork and exit (corresponding to `Type=forking`). We cannot know in advance which forking model the app uses.
In addition, some apps might launch detached instances of themselves at a later time (after a double fork), expecting the detached instance to survive the initial one when it exits

**Describe the solution you'd like**
A new `Type=any` (or `Type=cgroup` ?) that considers the service running as long as at least one process in the tree is running (doesn't rely on PID files or PID guessing as there is no specific main process).
The exit status of the service is that of the last process in the cgroup, so it will be considered to have crashed only if the last process crashed, not if any other process in the cgroup crashes.

**Describe alternatives you've considered**
Using `Type=simple` with default `KillMode` leads to the entire application being killed as soon as the parent process exits.
Changing to `KillMode=none` means we lose graceful shutdown
Using `Type=forking` means regular non-forking apps are never considered started

**Comment by unusual-thoughts** [comment]
renamed to `ExitType=main|cgroup`

**Comment by bluca** [comment]
Reopening - unfortunately https://github.com/systemd/systemd/pull/18782 had to be reverted to unblock the v249 release, as it was causing issues ( https://github.com/systemd/systemd/issues/19345 and https://github.com/systemd/systemd/pull/19385 ). Would be great to have a new PR that reintroduces the feature and fixes those issues as well
